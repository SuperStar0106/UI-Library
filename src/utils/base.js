import { injectGlobal, css } from 'styled-components'

import reset from './reset.styled'
import { fontFace } from './font.styled'

const baseResponsiveStyles = css`
  @media (max-width: 1200px) {
    html {
      font-size: 14px;
    }
  }

  @media (max-width: 1300px) and (max-height: 700px) {
    html {
      font-size: 14px;
    }
  }

  @media (max-width: 600px) {
    html {
      font-size: 16px;
    }
  }
`

const baseStyles = () => injectGlobal`
  ${reset}

  ${fontFace('Regular', 300)}
  ${fontFace('Medium', 400)}
  ${fontFace('Bold', 600)}
  ${fontFace('Heading', 700)}

  html {
    font-size: 16px;
    scroll-behavior: smooth;
  }

  body{
    font-family: 'HKCompakt';
    -webkit-font-smoothing: antialiased;
  }

  h1, h2, h3{
    line-height: 1.3;
  }

  h4, h5, h6, p, li{
    line-height: 1.4;
  }

  ::selection {
    background-color: #00C29A;
    color: #fff;
  }

  ${baseResponsiveStyles}
`

export default baseStyles
