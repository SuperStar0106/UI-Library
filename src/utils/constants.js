export const DEFAULT_FONT_SIZE = 50
export const DEFAULT_FONT_FAMILY = 'HKCompakt'
