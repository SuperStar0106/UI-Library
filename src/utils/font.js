import RegularWoff from '../../fonts/HKCompakt-Regular.woff'
import RegularTtf from '../../fonts/HKCompakt-Regular.ttf'

import MediumWoff from '../../fonts/HKCompakt-Medium.woff'
import MediumTtf from '../../fonts/HKCompakt-Medium.ttf'

import BoldWoff from '../../fonts/HKCompakt-Bold.woff'
import BoldTtf from '../../fonts/HKCompakt-Bold.ttf'

import HeadingWoff from '../../fonts/HKCompakt-Heading.woff'
import HeadingTtf from '../../fonts/HKCompakt-Heading.ttf'

export const fontFace = (version, weight) => {
  switch (version) {
    case 'Medium':
      return `
        @font-face{
          font-family: 'HKCompakt';
          src: url(${MediumWoff}) format('woff'),
               url(${MediumTtf}) format('truetype'),
          font-weight: ${weight};
        }
      `
    case 'Bold':
      return `
        @font-face{
          font-family: 'HKCompakt';
          src: url(${BoldWoff}) format('woff'),
               url(${BoldTtf}) format('truetype'),
          font-weight: ${weight};
        }
      `
    case 'Heading':
      return `
        @font-face{
          font-family: 'HKCompakt';
          src: url(${HeadingWoff}) format('woff'),
               url(${HeadingTtf}) format('truetype'),
          font-weight: ${weight};
        }
      `
    default:
      return `
        @font-face{
          font-family: 'HKCompakt';
          src: url(${RegularWoff}) format('woff'),
               url(${RegularTtf}) format('truetype'),
          font-weight: ${weight};
        }
      `
  }
}
