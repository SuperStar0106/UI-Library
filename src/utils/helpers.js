import { css } from 'styled-components'
import get from 'lodash.get'
import merge from 'lodash.merge'
import concat from 'lodash.concat'

import hexToRGB from './hexToRGB'

let helpers = {}

helpers.rgba = (...args) => {
  return props => {
    const opacity = args[args.length - 1]
    const path = args.slice(0, -1)
    const color = get(props, ['theme', 'color', ...path]) || args[0] // Handle rgba('#COFFEE', 0.3)

    return `rgba(${hexToRGB(color)}, ${opacity})`
  }
}

helpers.centeredContainerWidth = (...path) => {
  return props => {
    const basename = ['theme', 'centeredContainerWidth']
    const pathname = props.width || (path.length && [...path]) || 'lg'
    return get(props, concat(basename, pathname))
  }
}

helpers.textStyles = key => {
  return props => {
    const { fontFamily, fontSize, fontWeight, letterSpacing } = props.theme
    let defaultValues = {
      family: fontFamily.texts,
      size: 'inherit',
      weight: 'inherit',
      transform: 'none',
      spacing: 'none'
    }
    const value = get(props, ['theme', 'text', key])
    if (value) {
      const { family, size, weight, transform, spacing } = value
      defaultValues = merge(defaultValues, {
        family: fontFamily[family],
        size: fontSize[size],
        weight: fontWeight[weight],
        transform: transform,
        spacing: letterSpacing[spacing]
      })
    }
    return css`
      font-family: ${defaultValues.family};
      font-size: ${defaultValues.size};
      font-weight: ${defaultValues.weight};
      text-transform: ${defaultValues.transform};
      letter-spacing: ${defaultValues.spacing};
    `
  }
}

export const createHelpers = theme => {
  const defaultHelpers = Object.keys(theme).reduce((acc, key) => {
    return {
      ...acc,
      [key]: (...path) => props => {
        if (!props.theme) {
          props = { ...props, theme }
        }
        const value = get(props, ['theme', key, ...path])
        if (process.env.NODE_ENV === 'development' && value === undefined) {
          console.warn(`${key}[${[...path]}] is not available in this theme`)
        }

        return value
      }
    }
  }, {})

  return { ...defaultHelpers, ...helpers }
}

export const { centeredContainerWidth, rgba, textStyles } = helpers

export default createHelpers
