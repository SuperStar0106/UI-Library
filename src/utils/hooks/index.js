export * from './attributes'
export * from './eventListener'
export * from './refElement'
