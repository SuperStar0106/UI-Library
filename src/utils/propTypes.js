import PropTypes from 'prop-types'

export const reduxFormFieldPropTypes = {
  input: PropTypes.shape({
    name: PropTypes.string.isRequired
  }).isRequired,
  meta: PropTypes.shape({
    error: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    warnings: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
    submitFailed: PropTypes.bool.isRequired,
    touched: PropTypes.bool.isRequired
  })
}

export const formFieldPropTypes = {
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  required: PropTypes.bool,
  autoFocus: PropTypes.bool
}

export const formFieldDefaultProps = {
  classList: [],
  inlineHint: null,
  label: null,
  placeholder: null,
  required: false,
  autoFocus: false
}
