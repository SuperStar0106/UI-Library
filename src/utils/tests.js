import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import StaticRouter from 'react-router-dom/StaticRouter'
import 'jest-styled-components'

import theme from '../theme/core'

export const render = children => {
  return renderer.create(
    <ThemeProvider theme={theme}>
      <StaticRouter context={{}}>{children}</StaticRouter>
    </ThemeProvider>
  )
}
