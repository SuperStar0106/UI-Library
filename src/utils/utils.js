import { css } from 'styled-components'
import { breakpoint, boxShadow, transition, rgba } from './helpers'

// Custom scrollbar mixin
export const scrollbar = (width, thumbColor, trackColor) => {
  return `
    &::-webkit-scrollbar {
        width:  ${width};
        height: ${width};
    }

    &::-webkit-scrollbar-thumb {
        background: ${thumbColor};
        border-radius: 10px;
    }

    &::-webkit-scrollbar-track {
        background: ${trackColor};
        border-radius: 10px;
    }

    & {
      scrollbar-face-color: ${thumbColor};
      scrollbar-track-color: ${trackColor};
    }
  `
}

export const getBackgroundImage = image => {
  return image ? css`background-image: url('${image}');` : null
}

export const media = {
  print: (...args) => css`
    @media print {
      ${css(...args)};
    }
  `,
  mobile: (...args) => css`
    @media (max-width: ${breakpoint('mobile')}) {
      ${css(...args)};
    }
  `,
  tablet: (...args) => css`
    @media (max-width: ${breakpoint('tablet')}) {
      ${css(...args)};
    }
  `,
  smallscreen: (...args) => css`
    @media (max-width: ${breakpoint('smallscreen')}) {
      ${css(...args)};
    }
  `,
  mediumscreen: (...args) => css`
    @media (max-width: ${breakpoint('mediumscreen')}) {
      ${css(...args)};
    }
  `,
  widescreen: (...args) => css`
    @media (max-width: ${breakpoint('widescreen')}) {
      ${css(...args)};
    }
  `
}

export const overflowEllipsis = css`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`

export const backgroundCover = css`
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
`

export const bannerMask = css`
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;

  &::after {
    position: absolute;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%;
    background-color: ${rgba('black', 0.5)};
    content: ' ';
  }
`

export const printHiddenStyles = css`
  ${media.print`
    display: none;
  `};
`

export const hoverTransformStyles = css`
  transition: ${transition('md')};

  &:hover {
    transform: translate3d(0, -5px, 0);
    box-shadow: ${boxShadow('articleThumbHover')};
  }
`
