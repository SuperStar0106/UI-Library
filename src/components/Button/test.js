import React from 'react'

import { render } from '../../utils/tests'

import { Button } from './index'

const content = 'Jungle'

test('<Button> renders correctly', () => {
  const { container } = render(<Button>{content}</Button>)
  expect(container).toHaveTextContent(content)
})
