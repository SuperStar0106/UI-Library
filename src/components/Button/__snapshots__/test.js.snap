// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
.c0 {
  color: #FAFAFA;
  font-size: 0.6875rem;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  background-color: #1B74B5;
  border-color: #1B74B5;
  border-radius: 4px;
  height: 2.25rem;
  padding: 0 0.75rem;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
  border-width: 1px;
  border-style: solid;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition: 300ms ease;
  transition: 300ms ease;
}

.c0:hover {
  box-shadow: 1px 2px 4px 0 rgba(0,0,0,0.05);
}

.c0:focus {
  box-shadow: 1px 2px 4px 0 rgba(0,0,0,0.05);
}

.c0 > *:last-child {
  margin-left: 0.625rem;
}

.c0::before {
  background: rgba(250,250,250,0.05);
}

.c0:active {
  -webkit-transform: translateY(2px);
  -ms-transform: translateY(2px);
  transform: translateY(2px);
}

.c0:hover,
.c0:focus {
  background-color: #208DDB;
  border-color: #208DDB;
}

.c0[disabled] {
  color: #828282;
  font-size: 0.6875rem;
  font-weight: 600;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  background-color: #CCCCCC;
  border-color: #CCCCCC;
  border-radius: 4px;
  pointer-events: none;
}

.c0[disabled]:hover {
  box-shadow: 1px 2px 4px 0 rgba(0,0,0,0.05);
}

.c0[disabled]:focus {
  box-shadow: 1px 2px 4px 0 rgba(0,0,0,0.05);
}

<button
  className="c0"
  size="auto"
>
  Test button
</button>
`;
